pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -676,30 +590,20 @@ def _set_snopt_options(self, iPrint, iSumm, cw, iw, rw):
         Set all the options into SNOPT that have been assigned
         by the user
         """
-
         # Set Options from the local options dictionary
         # ---------------------------------------------
         inform = np.array([-1], np.intc)
-        for name, value in self.set_options.items():
-            if name == "iprint" or name == "isumm":
+        for name, value in self.options.items():
+            # these do not get set using snset
+            if name in self.specialOptions or name in self.pythonOptions:
                 continue
 
             if isinstance(value, str):
-                if name == "start":
-                    if value == "Cold":
-                        snopt.snset("Cold start", iPrint, iSumm, inform, cw, iw, rw)
-                    elif value == "Warm":
-                        snopt.snset("Warm start", iPrint, iSumm, inform, cw, iw, rw)
-                elif name == "problem type":
-                    if value == "Minimize":
-                        snopt.snset("Minimize", iPrint, iSumm, inform, cw, iw, rw)
-                    elif value == "Maximize":
-                        snopt.snset("Maximize", iPrint, iSumm, inform, cw, iw, rw)
-                    elif value == "Feasible point":
-                        snopt.snset("Feasible point", iPrint, iSumm, inform, cw, iw, rw)
-                elif name == "print file":
+                if name == "Problem Type":
+                    snopt.snset(value, iPrint, iSumm, inform, cw, iw, rw)
+                elif name == "Print file":
```

Comment:
Double checking again: the `Print file`,  `Summary file`,  .. are not passed directly to the fortran layer as options, but they are passed directly to SNOPT subroutines right? The outfiles are set up in the `__call__` method above if I get it right

Reply:
Yep the openunit/closeunit actually handles the file name. We just pass the file number (`iSumm` and `iPrint`) to SNOPT, and that's what this section does.

Reply:
`iSumm` and `iPrint` are handled by Fortran at every optimizer iteration? I'll check the manual in case

Reply:
Not at every iteration, but used for setting options. If you check `pySNOPT` you can see where `iSumm` and `iPrint` are passed to SNOPT.